src/test/java/uk/gov/pay/directdebit/payers/services/PayerServiceTest.java

```diff
@@ -64,10 +67,22 @@ public void setUp() {
     }
 
     @Test
-    public void shouldStoreAPayerWhenReceivingCreatePayerRequest() {
-        service.create(paymentRequestExternalId, gatewayAccount.getExternalId(), createPayerRequest);
+    public void shouldCreateAndStoreAPayerWhenReceivingCreatePayerRequest() {
+        service.createOrUpdatePayer(paymentRequestExternalId, gatewayAccount, createPayerRequest);
         verify(mockedPayerDao).insert(payer);
         verify(mockedTransactionService).payerCreatedFor(transaction);
+        verify(mockedTransactionService, never()).payerEditedFor(transaction);
+    }
+    @Test
+    public void shouldUpdateAndStoreAPayerWhenReceivingCreatePayerRequest_ifAPayerAlreadyExists() {
+        Payer payer = PayerFixture.aPayerFixture()
```

Comment:
The variable shadowing makes things a little unclear here: maybe call it `originalPayer` or something?